src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -1202,15 +1202,15 @@ public void M3()
 
             var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef });
             comp.VerifyDiagnostics(
-                // (14,20): error CS0610: Field or property cannot be of type 'ArgIterator'
-                //     public static (ArgIterator, ArgIterator) M2()
-                Diagnostic(ErrorCode.ERR_FieldCantBeRefAny, "ArgIterator").WithArguments("System.ArgIterator").WithLocation(14, 20),
-                // (14,33): error CS0610: Field or property cannot be of type 'ArgIterator'
-                //     public static (ArgIterator, ArgIterator) M2()
-                Diagnostic(ErrorCode.ERR_FieldCantBeRefAny, "ArgIterator").WithArguments("System.ArgIterator").WithLocation(14, 33),
                 // (18,29): error CS1601: Cannot make reference to variable of type 'ArgIterator'
                 //     public void Deconstruct(out ArgIterator a, out int b)
                 Diagnostic(ErrorCode.ERR_MethodArgCantBeRefAny, "out ArgIterator a").WithArguments("System.ArgIterator").WithLocation(18, 29),
+                // (14,46): error CS0306: The type 'ArgIterator' may not be used as a type argument
+                //     public static (ArgIterator, ArgIterator) M2()
+                Diagnostic(ErrorCode.ERR_BadTypeArgument, "M2").WithArguments("System.ArgIterator").WithLocation(14, 46),
+                // (14,46): error CS0306: The type 'ArgIterator' may not be used as a type argument
```

Comment:
The locations for reporting errors on return type of `M2` still point to the word `M2`, rather than the tuple elements.

---
In reply to: [91575656](https://github.com/dotnet/roslyn/pull/15641#discussion_r91575656) [](ancestors = 91575656)